### PR TITLE
Add bucket name to db

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**
- add bucket name to db
- add bucketName filter to completedJobs endpoint
- add bucketName filter to revisionJobsInfo endpoint
- add bucketName to JobsInfoData

### CODE REVIEW

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java [hunk 1]
```diff
@@ -766,6 +767,9 @@ private void handleJobAttributes(Job commonPropertiesHolder, Map<String, String>
             } else if (XMLAttributes.JOB_PROJECT_NAME.matches(attributeName)) {
                 commonPropertiesHolder.setProjectName(replace(attributeValue,
                                                               commonPropertiesHolder.getVariablesAsReplacementMap()));
+            } else if (XMLAttributes.JOB_BUCKET_NAME.matches(attributeName)) {
```
Comment (fviale) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java:770]: why did you change the xml schema ???

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobData.java [hunk 2]
```diff
@@ -616,6 +621,15 @@ public void setProjectName(String projectName) {
         this.projectName = projectName;
     }
 
+    @Column(name = "BUCKET_NAME", updatable = false)
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobData.java:624]: please add an index to the BUCKET_NAME column

**Comment by oanaBiancaSchiopu** [comment]
> 

yes I will add it

**Comment by oanaBiancaSchiopu** [comment]
retest this please

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java (hunk 1 above)
Comment (fviale) [reply scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java:770]: As you set the bucket name in LiveJobs, StaxJobFactory should not be modified.

**Comment by oanaBiancaSchiopu** [comment]
I prefer to include the bucketName filter of revisionJobsInfo endpoint and other endpoints in a different PR, as there would be harder to review a big number of changed files.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobData.java [hunk 3]
```diff
@@ -156,7 +156,8 @@
                                       @Index(name = "JOB_DATA_PARENT_RANK_ID", columnList = "PARENT_JOB_ID,STATUS_RANK DESC,ID DESC"),
                                       @Index(name = "JOB_DATA_OWNER_PARENT_RANK_ID", columnList = "OWNER,PARENT_JOB_ID,STATUS_RANK DESC,ID DESC"),
                                       @Index(name = "JOB_DATA_FAILED_TASKS", columnList = "FAILED_TASKS"),
-                                      @Index(name = "JOB_DATA_FAULTY_TASKS", columnList = "FAULTY_TASKS") })
+                                      @Index(name = "JOB_DATA_FAULTY_TASKS", columnList = "FAULTY_TASKS"),
+                                      @Index(name = "JOB_DATA_BUCKET_NAME", columnList = "BUCKET_NAME"), })
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobData.java:160]: could you also add an index on project name ?

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java [hunk 4]
```diff
@@ -190,6 +190,8 @@ List<UserJobData> jobsInfoListPost(@HeaderParam("sessionid") String sessionId, L
      *             Include only jobs with a name that starts with jobName (case in-sensitive)
      * @param projectName
      *             Include only jobs with a project name that starts with projectName (case in-sensitive)
+     * @param bucketName
```
Comment (fviale) [inline rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java:193]: actually all these filters (jobName, projectName, bucketName and userName) are case sensitive

File: scheduler/scheduler-server/src/test/java/functionaltests/db/schedulerdb/TestLoadJobsPagination.java [hunk 5]
```diff
@@ -65,6 +65,11 @@ private TaskFlowJob createJob(String name, JobPriority priority, String projectN
         } else {
             job.setProjectName(this.getClass().getSimpleName() + " project");
         }
+        if (bucketName != null) {
+            job.setBucketName(bucketName);
+        } else {
```
Comment (fviale) [inline scheduler/scheduler-server/src/test/java/functionaltests/db/schedulerdb/TestLoadJobsPagination.java:70]: why do you **have to** set a non-null bucket name if not provided in the parameter in this test ?

File: rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java [hunk 6]
```diff
@@ -338,6 +338,7 @@ public Page<JobInfo> getJobs(int index, int range, JobFilterCriteria criteria,
                                                                                                            criteria.isChildJobs(),
                                                                                                            criteria.getJobName(),
                                                                                                            criteria.getProjectName(),
+                                                                                                           criteria.getBucketName(),
```
Comment (fviale) [inline rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java:341]: in the method getJobInfo(..) of this class you should add

jobInfoImpl.setBucketName(jobInfoData.getBucketName());

Please do so for other recently added attributes if not already.

**Review by fviale — APPROVED** [review]
LGTM

**Review by mklkun — APPROVED** [review]
